lasagne/layers/noise.py

```diff
@@ -211,3 +212,88 @@ def get_output_for(self, input, deterministic=False, **kwargs):
             return input + self._srng.normal(input.shape,
                                              avg=0.0,
                                              std=self.sigma)
+
+
+class AlphaDropoutLayer(Layer):
+    """Dropout layer.
+
+    Sets values to zero with probability p. Will also converge the remaining
+    neurons to mean 0 and a variance of 1 unit. See notes for disabling dropout
+    during testing.
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        the layer feeding into this layer, or the expected input shape
+    p : float or scalar tensor
+        The probability of setting a value to zero
+    rescale : bool
+        If ``True`` (the default), scale the input by ``1 / (1 - p)`` when
+        dropout is enabled, to keep the expected output mean the same.
+    shared_axes : tuple of int
+        Axes to share the dropout mask over. By default, each value can be
+        dropped individually. ``shared_axes=(0,)`` uses the same mask across
+        the batch. ``shared_axes=(2, 3)`` uses the same mask across the
+        spatial dimensions of 2D feature maps.
+
+    Notes
+    -----
+    The alpha dropout layer is a regularizer that randomly sets input values to
+    zero and also applies a normalization function to bring the mean to 0
+    and the variance to 1; see [1]_for why this might improve
+    generalization. The behaviour of the layer depends on the ``deterministic``
+    keyword argument passed to :func:`lasagne.layers.get_output`. If ``True``,
+    the layer behaves deterministically, and passes on the input unchanged. If
+    ``False`` or not specified, dropout (and possibly scaling) is enabled.
+    Usually, you would use ``deterministic=False`` at train time and
+    ``deterministic=True`` at test time.
+    References
+    ----------
+    .. [1] Klambauer, G., Unterthiner, T., Mayr, A., Hochreiter, S. (2017):
+           Self-Normalizing Neural Networks. arXiv preprint: 1706.02515
+    """
+
+    def __init__(self, incoming, p=0.1, rescale=True, shared_axes=(),
+                 **kwargs):
+        """Class initialization."""
+        super(AlphaDropoutLayer, self).__init__(incoming, **kwargs)
+        self._srng = RandomStreams()
+        self.p = p
+        self.alpha = -1.7580993408473766
```

Comment:
Oh, and we should use all available digits, because we can. And maybe call it differently, since it's called alpha prime in the paper, computed as minus lambda times alpha? Not sure.

The default value would be `-1.0507009873554804934193349852946 * 1.6732632423543772848170429916717`.